### DISCUSSION TITLE
Allow Read but not List?

### DISCUSSION THREAD

**Discussion by lsei** [supabase/supabase#3785] [Questions]
I'm currently implementing access to user profiles for logged in users via an additional users table in the public schema as has been recommended.

Is there any way to allow api calls for a single result at a time given a userid, while disallowing retrieval of all profiles at once (without knowing their ids?)

I haven't seen any examples of this using RLS / PostgREST yet.

**Comment by steve-chavez** [comment]
This doesn't work yet but I was thinking of adding this capability:

```sql
CREATE POLICY "Disallow listing" ON profiles
USING (
  current_setting('request.params', true)::json->>'userid.eq' is not null
  -- or the above could be wrapped as
  -- pgrest.is_eq_filtered('userid')
);
```

This would basically enforce the presence of an [eq](https://supabase.io/docs/reference/javascript/eq) filter on `userid`( `WHERE userid = <val>`).

Would that work for you? WDYT?

(The above depends on https://github.com/PostgREST/postgrest/pull/1710)

**Reply by lsei** [reply]
Hi Steve, Thanks for this. 

Exposing the query parameters would be really helpful. Perhaps even just as single json object that someone can write a function around would be a great start if a per-parameter GUC adds a lot of overhead. 

A specific `is_eq_filtered` function seems sensible for the single result use case. 

Are the current `pgrest.` methods documented somewhere?

**Reply by steve-chavez** [reply]
@lsei Thanks for the feedback!

> Are the current pgrest. methods documented somewhere?

They're not available yet but basically they'd be wrappers around the query parameters' GUCs.

> Is there any way to allow api calls for a single result at a time given a userid, while disallowing retrieval of all profiles at once (without knowing their ids?)

For now you could work around this with an [rpc](https://supabase.io/docs/reference/javascript/rpc). So you'd put your profiles table on a `private` schema and only expose a `get_profile` function through your API by adding it to your `public` schema. This `get_profile` function would internally do a RAISE(like on https://github.com/supabase/supabase/discussions/3732#discussioncomment-1579200) if a `userid` argument is not present.
